src/wallet/rpcwallet.cpp

```diff
@@ -3929,6 +3930,100 @@ UniValue z_setmigration(const UniValue& params, bool fHelp) {
+            "  \"finalized_migrated_amount\": nnn.n,       (numeric) The total amount of finalized " + CURRENCY_UNIT + " \n"
+            "  \"finalized_migration_transactions\": nnn,  (numeric) The number of migration transactions involving this wallet\n"
+            "  \"time_started\": ttt,                      (numeric, optional) The block time of the first migration transaction\n"
+            "  \"migration_txids\": [txids]                (json array of strings) An array of all migration txids involving this wallet\n"
+            "}\n"
+        );
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+    UniValue migrationStatus(UniValue::VOBJ);
+    migrationStatus.push_back(Pair("enabled", pwalletMain->fSaplingMigrationEnabled));
+    //  The "destination_address" field MAY be omitted if the "-migrationaddress"
+    // parameter is not set and no default address has yet been generated.
+    // Note: The following function may return the default address even if it has not been added to the wallet
+    auto destinationAddress = AsyncRPCOperation_saplingmigration::getMigrationDestAddress(pwalletMain->GetHDSeedForRPC());
+    migrationStatus.push_back(Pair("destination_address", EncodePaymentAddress(destinationAddress)));
+    //  The values of "unmigrated_amount" and "migrated_amount" MUST take into
+    // account failed transactions, that were not mined within their expiration
+    // height.
+    {
+        std::vector<CSproutNotePlaintextEntry> sproutEntries;
+        std::vector<SaplingNoteEntry> saplingEntries;
+        pwalletMain->GetFilteredNotes(sproutEntries, saplingEntries, "", 1);
+        CAmount unmigratedAmount = 0;
+        for (const auto& sproutEntry : sproutEntries) {
+            unmigratedAmount += sproutEntry.plaintext.value();
+        }
+        migrationStatus.push_back(Pair("unmigrated_amount", FormatMoney(unmigratedAmount)));
+    }
+    //  "migration_txids" is a list of strings representing transaction IDs of all
+    // known migration transactions involving this wallet, as lowercase hexadecimal
+    // in RPC byte order.
+    UniValue migrationTxids(UniValue::VARR);
+    int currentHeight = chainActive.Height();
+    CAmount unfinalizedMigratedAmount = 0;
+    CAmount finalizedMigratedAmount = 0;
+    int numFinalizedMigrationTxs = 0;
+    uint64_t timeStarted = 0;
+    for (const auto& txPair : pwalletMain->mapWallet) {
+        CWalletTx tx = txPair.second;
+        // A given transaction is defined as a migration transaction iff it has:
+        // * one or more Sprout JoinSplits with nonzero vpub_new field; and
+        // * no Sapling Spends, and;
+        // * one or more Sapling Outputs.
+        if (tx.vjoinsplit.size() > 0 && tx.vShieldedSpend.empty() && tx.vShieldedOutput.size() > 0) {
+            CAmount migrationAmount = 0;
+            for (const auto& js : tx.vjoinsplit) {
+                migrationAmount += js.vpub_new;
+            }
```

Comment:
I think it makes sense to update the spec to include `vin.empty() && vout.empty()` in the definitions of migration transactions. Also I think it is correct to to use `-valueBalance` as opposed to `js.vpub_new`, so I will update that. Regarding `js.vpub_old`, it might make sense to change the definition to be `one or more Sprout JoinSplits all zero vpub_old fields` rather than `one or more Sprout JoinSplits with nonzero vpub_new field`, but would require more discussion so I am leaving it is for the time being.

Reply:
One thing to note is that the changes regarding `vin`, `vout`, and the proposed change regarding `vpub_old` will only effect manually created migration transactions which may be considered unsupported as there is no way to create them using a standard zcashd node. For that reason I do not think it is super pressing to switch to using `vpub_old` rather than `vpub_new`, but it is still something to consider.

Reply:
I've thought about this and continue to believe that the current definition in the ZIP is the simplest that satisfies the needed criteria.

Reply:
In particular, what would be the rationale of enforcing that `vpub_old`s are zero, but not also enforcing that there are no transparent inputs? Do we also enforce that there are no transparent outputs? Where do we stop?

Reply:
Agreed that it is correct to use `-valueBalance`. I think that gives a sensible result independent of the migration transaction criteria; what we're saying is "there are transactions that in total transferred X into Sapling, at least some of it coming from Sprout (and for transactions created by zcashd, exactly X will be from Sprout)".

Reply:
> Where do we stop?

Good point. It is a bit of a slippery slope. I will change it back to not look at vin/vout.